c10/core/StorageImpl.h

```diff
@@ -210,11 +211,22 @@ struct C10_API StorageImpl final : public c10::intrusive_ptr_target {
     numel_ = capacity / data_type_.itemsize();
   }
 
+  void set_received_cuda(bool received_cuda) {
+    received_cuda_ = received_cuda;
+  }
+
+  bool received_cuda() {
+    return received_cuda_;
+  }
+
  private:
   caffe2::TypeMeta data_type_;
   DataPtr data_ptr_;
   int64_t numel_;
   bool resizable_;
+  // Identifies that Storage was received from another process and doesn't have
+  // local to process cuda memory allocation
+  bool received_cuda_;
```

Comment:
nit: It might be worth noting that this attribute is *solely* used to prevent a user from trying to resend a tensor to another process. This is mostly to encourage people not to use this boolean for other things without thinking about it.